### PR TITLE
ARROW-1132: [Python] Unable to write pandas DataFrame w/MultiIndex containing duplicate values to parquet

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

File: python/pyarrow/tests/test_parquet.py [hunk 1]
```diff
@@ -32,29 +32,21 @@
 
 import pandas.util.testing as tm
 
+pq = pytest.importorskip('pyarrow.parquet')
```
Comment (wesm) [inline python/pyarrow/tests/test_parquet.py]: Please revert this. See https://github.com/apache/arrow/commit/8f2b44b897b7083ee2a296c70397dc2d7d21d95e#diff-3a10a971f558573678baea521d62790a

Comment (wesm) [inline python/pyarrow/tests/test_parquet.py]: The problem is if the extension is built but `pyarrow.parquet` fails to import due to a dynamic linking error, the whole module gets ignored

Comment (cpcloud) [inline python/pyarrow/tests/test_parquet.py]: Done.

File: cpp/src/arrow/python/pandas_convert.cc [hunk 2]
```diff
@@ -1310,7 +1310,6 @@ inline Status ConvertNulls(const ChunkedArray& data, PyObject** out_values) {
   for (int c = 0; c < data.num_chunks(); c++) {
     std::shared_ptr<Array> arr = data.chunk(c);
 
-    const bool has_nulls = data.null_count() > 0;
```
Comment (cpcloud) [inline cpp/src/arrow/python/pandas_convert.cc]: I'll remove this after #774 is merged.

**Comment by cpcloud** [comment]
Added another test to confirm that writing works. The parquet writing ended up actually being a red herring for a deeper problem which was that we couldn't handle duplicate multiindex levels at all. The problem is fixed in this patch.

**Comment by cpcloud** [comment]
@wesm passing on appveyor: https://ci.appveyor.com/project/cpcloud/arrow/build/1.0.184

**Review by wesm — APPROVED** [review]
+1. Thank you!

**Comment by bmabey** [comment]
Thanks @cpcloud for fixing this so quickly!

When would you expect this fix to make it into a release/conda build?

**Comment by wesm** [comment]
The conda-forge packages can be updated whenever; help would be appreciated with that. We have a bunch of patches in flight so it might be worth waiting a couple days before starting the update process on conda-forge (for arrow-cpp/parquet-cpp/pyarrow)

**Comment by bmabey** [comment]
> The conda-forge packages can be updated whenever; help would be appreciated with that.

Are you suggesting a `0.5-SNAPSHOT` or similar build for conda-forge? (I would vote for something along the lines of `0.5-gitref`)

Is the release procedure simply opening up a PR for the feedstock after updating the [recipe](https://github.com/conda-forge/pyarrow-feedstock/blob/master/recipe/meta.yaml)? If so I'm happy to do that. I'm okay waiting till next week to catch a few more patches.

**Comment by wesm** [comment]
Yeah, the compliant version number would be 0.5.0.pre (with a conda package build number that we can increment after that). We'll have to be careful with version pinning so that people can still install 0.4.1 without getting the newer packages (the upgrade from 0.3.0 to 0.4.0 was not as graceful as planned)

**Comment by bmabey** [comment]
@wesm is the most recent commit a good option to make a release? 

This is how I've edited the feedstock so far, does it look good? https://github.com/bmabey/pyarrow-feedstock/commit/53b5e206f2bbdba508b7dac8576374de5bd89955

**Comment by wesm** [comment]
It's a bit tricky because pyarrow depends on arrow-cpp and parquet-cpp, both of which have just undergone some changes to their build systems, so those recipes also need to be updated. 

Once https://github.com/apache/parquet-cpp/pull/364 is merged I can start updating the arrow-cpp and parquet-cpp feedstocks, I will keep you posted

**Comment by wesm** [comment]
@bmabey sorry I didn't finish the updates -- we are close to cutting a 0.5.0 release candidate so I will try to make a 0.5.0.pre release on conda-forge before the final release goes out, also to make sure nothing broke in our packaging since 0.4.1 (since we moved a bunch of code around in the package toolchain)

**Comment by wesm** [comment]
Should do this conda-forge update in the next few days if all goes well

**Pull request closed without merging**